vllm/distributed/device_communicators/shm_broadcast.py

```diff
@@ -21,6 +23,13 @@
 
 logger = init_logger(__name__)
 
+# We prefer to use os.sched_yield as it results in tighter polling loops,
+# measured to be around 3e-7 seconds. However on python < 3.11.1,
+# os.sched_yield() does not release the GIL, so we fall back to time.sleep(0)
+USE_SCHED_YIELD = False
+if sys.version_info[:3] >= (3, 11, 1):
+    USE_SCHED_YIELD = True
```

Comment:
You could keep the logic about which method to use here, as well with something like this ...

```suggestion
# We prefer to use os.sched_yield as it results in tighter polling loops,
# measured to be around 3e-7 seconds. However on python < 3.11.1,
# os.sched_yield() does not release the GIL, so we fall back to time.sleep(0)
def sched_yield():
    if sys.version_info[:3] >= (3, 11, 1):
        os.sched_yield()
    else:
        time.sleep(0)
```

Reply:
Agree with having a separate function but we might still want to keep the constant to avoid slicing/parsing every time.

nit simplification:
```python
USE_SCHED_YIELD = sys.version_info[:3] >= (3, 11, 1)
```

Also I think the fix was backported to 3.10, we could possibly check for that too.

Reply:
Nice, you're right - needs 3.11.1 or 3.10.8
https://docs.python.org/3.11/whatsnew/changelog.html
https://docs.python.org/3.10/whatsnew/changelog.html